modules/fluid_properties/src/fluidproperties/TabulatedFluidProperties.C

```diff
@@ -1104,38 +1135,92 @@ TabulatedFluidProperties::s_from_h_p(Real h, Real pressure) const
+      file_out << "\n";
+
+      // Write out the fluid property data
+      for (unsigned int p = 0; p < _num_p; ++p)
+        for (unsigned int t = 0; t < _num_T; ++t)
+        {
+          file_out << _pressure[p] << ", " << _temperature[t];
+          for (std::size_t i = 0; i < _properties.size(); ++i)
+            file_out << ", " << _properties[i][p * _num_T + t];
+          file_out << "\n";
+        }
 
-    // Write out column names
-    file_out << "pressure, temperature";
-    for (std::size_t i = 0; i < _interpolated_properties.size(); ++i)
-      file_out << ", " << _interpolated_properties[i];
-    file_out << "\n";
+      file_out << std::flush;
+      file_out.close();
+    }
 
-    // Write out the fluid property data
-    for (unsigned int p = 0; p < _num_p; ++p)
-      for (unsigned int t = 0; t < _num_T; ++t)
-      {
-        file_out << _pressure[p] << ", " << _temperature[t];
-        for (std::size_t i = 0; i < _properties.size(); ++i)
-          file_out << ", " << _properties[i][p * _num_T + t];
-        file_out << "\n";
-      }
+    // Write out the (v,e) to (p,T) conversions
+    if (_construct_pT_from_ve)
+    {
+      const auto file_name_ve = std::regex_replace(file_name, std::regex(".csv"), "_ve.csv");
+      MooseUtils::checkFileWriteable(file_name_ve);
+      std::ofstream file_out(file_name_ve.c_str());
+
+      // Write out column names
+      file_out << "specific_volume, internal_energy, pressure, temperature";
+      for (std::size_t i = 0; i < _interpolated_properties.size(); ++i)
+        if (_interpolated_properties[i] != "internal_energy")
+          file_out << ", " << _interpolated_properties[i];
+      file_out << "\n";
+
+      // Write out the fluid property data
+      for (const auto v : make_range(_num_v))
+        for (const auto e : make_range(_num_e))
+        {
+          const auto v_val = _specific_volume[v];
+          const auto e_val = _internal_energy[e];
+          const auto pressure = _p_from_v_e_ipol->sample(v_val, e_val);
+          const auto temperature = _T_from_v_e_ipol->sample(v_val, e_val);
+          file_out << v_val << ", " << e_val << ", " << pressure << ", " << temperature;
+          for (std::size_t i = 0; i < _properties.size(); ++i)
+          {
+            if (i == _density_idx)
+              file_out << 1 / v_val << ", ";
+            else if (i == _enthalpy_idx)
+              file_out << h_from_p_T(pressure, temperature) << ", ";
+            else if (i == _viscosity_idx)
+              file_out << mu_from_v_e(v_val, e_val) << ", ";
+            else if (i == _k_idx)
+              file_out << k_from_v_e(v_val, e_val) << ", ";
+            else if (i == _c_idx)
+              file_out << c_from_v_e(v_val, e_val) << ", ";
+            else if (i == _cp_idx)
+              file_out << cp_from_v_e(v_val, e_val) << ", ";
+            else if (i == _cv_idx)
+              file_out << cv_from_v_e(v_val, e_val) << ", ";
```

Comment:
```suggestion
            else if (i == _cv_idx)
              file_out << cv_from_v_e(v_val, e_val) << ", ";
            else if (i == _cp_idx)
              file_out << cp_from_v_e(v_val, e_val) << ", ";
```
I think it should be the reverse order.

Reply:
the 'if' will take care of that